core/src/main/java/hudson/model/labels/LabelAtom.java

```diff
@@ -106,6 +106,8 @@ public String getExpression() {
         return Collections.unmodifiableList(actions);
     }
 
+    // TODO implement addAction, addOrReplaceAction, removeAction, removeActions, replaceActions
```

Comment:
`addAction` has always been there, but I think would not have worked for years, since before #1059. They would have started to work as of #2608 but stopped working again almost immediately as of #2624. So I believe there is no backward compatibility issue with these.